lib/bracket-matcher-view.js

```diff
@@ -216,6 +221,28 @@ class BracketMatcherView {
     return result
   }
 
+  findMatchingTagsWithSyntaxTree () {
+    const currentPosition = this.editor.getCursorBufferPosition()
+    let startRange, endRange
+    const node = this.editor.buffer.getLanguageMode().getSyntaxNodeAtPosition(currentPosition, node => {
+      if (node.type.includes('element') && node.childCount > 1) {
+        const {firstChild, lastChild} = node
+        if (firstChild.childCount > 2 && firstChild.firstChild.type === '<') {
+          startRange = firstChild.child(1).range
+          if (lastChild.childCount > 2) {
+            if (lastChild.firstChild.type === '</') {
+              endRange = lastChild.child(1).range
+            } else if (lastChild.firstChild.type === '<' && lastChild.child(1).type === '/') {
+              endRange = lastChild.child(2).range
+            }
+          }
+          return true
+        }
+      }
+    })
+    if (startRange && endRange) return {startRange, endRange}
+  }
```

Comment:
This one method takes the place of the [the `TagFinder` class](https://github.com/atom/bracket-matcher/blob/24f4d0282e6e6ccb5fd4fbc70939e69a6a9ff21b/lib/tag-finder.js).